### ISSUE TITLE
Updating or Adding profiles have strange behavior for merging indices 

### ISSUE THREAD

**Issue by lettergram** [capitalone/DataProfiler#167]
Example:
```
df_series1 = [None, 1, 2, 3]
df_series2 = [0, None, 2, 3]
profile1 = StructuredDataProfile(df_series1)
profile2 = StructuredDataProfile(df_series2)
profile3 = profile1 + profile2
profile3.profile.null_types_index['None'] # Will be [0, 1]
```

This should be something such as [0, 5] (basically, starting the index location at the end of the first profile). This could be different if the index locations were provided and there was no overlap.

**Comment by ChrisWallace2020** [comment]
I'll tackle this starting 4/26

**Comment by lettergram** [comment]
To preface, I'm not confident which approach is better. Simply pointing out the implications of a couple of approaches:

### Example 1 (current):
Use Index locations, union dicts
```
data1 = Data(file1)
#  [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [0, 1, 2, None, None, None, 6, 7, 8, 9]]

data2 = Data(file2) 
 #   [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
 #    [None, None, None, 13, 14, 15, 16, 17, 18, 19]]
 
profile1 = profiler(data1) # null_list [3, 4, 5]
profile2 = profiler(data2) # null_list [0, 1, 2]

profile3 = profile1 + profile2
profile3.null_list # [0, 1, 2, 3, 4, 5]
profile3.sample_size # 20
profile3.null_ratio # 0.25

data3 = Data(file3)
#    [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [21, 22, 23, 24, 25, None, None, None, None]] 

profile3.update(data3) # null_list [6, 7, 8, 9]
profile3.null_list # [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] <---- Is this accurate?
profile3.sample_size # 30
profile3.null_ratio # 0.3333

data4 = Data(file4)
#    [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [21, 22, 23, 24, 25, None, None, None, None]] 

profile3.update(data4) # initial null_list [6, 7, 8, 9]
profile3.null_list # [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] <---- Is this accurate?
profile3.sample_size # 40
profile3.null_ratio # 0.25 <----- Not accurate
```


### Example 2:
Use Index locations, not union dicts
```
data1 = Data(file1)
#  [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [0, 1, 2, None, None, None, 6, 7, 8, 9]]

data2 = Data(file2) 
 #   [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
 #    [None, None, None, 13, 14, 15, 16, 17, 18, 19]]
 
profile1 = profiler(data1) # null_list [3, 4, 5]
profile2 = profiler(data2) # null_list [0, 1, 2]

profile3 = profile1 + profile2
profile3.null_list # [0, 1, 2, 3, 4, 5]
profile3.sample_size # 20
profile3.null_ratio # 0.25

data3 = Data(file3)
#    [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [21, 22, 23, 24, 25, None, None, None, None]] 

profile3.update(data3) # null_list [6, 7, 8, 9]
profile3.null_list # [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] 
profile3.sample_size # 30
profile3.null_ratio # 0.3333

data4 = Data(file4)
#    [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [21, 22, 23, 24, 25, None, None, None, None]] 

profile3.update(data4) # initial null_list [6, 7, 8, 9], updated [16, 17, 18, 19]
profile3.null_list # [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 16, 17, 18, 19] <---- Is this accurate?
profile3.sample_size # 40
profile3.null_ratio # 0.35 <----- Not accurate
```


### Example 3:
Utilize iloc, not union dicts
```
data1 = Data(file1)
#  [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [0, 1, 2, None, None, None, 6, 7, 8, 9]]

data2 = Data(file2) 
 #   [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
 #    [None, None, None, 13, 14, 15, 16, 17, 18, 19]]
 
profile1 = profiler(data1) # null_list [3, 4, 5]
profile2 = profiler(data2) # null_list [0, 1, 2]

profile3 = profile1 + profile2
profile3.null_list # [0, 1, 2, 13, 14, 15]
profile3.sample_size # 20
profile3.null_ratio # 0.25

data3 = Data(file3)
#    [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [21, 22, 23, 24, 25, None, None, None, None]] 

profile3.update(data3) # null_list [6, 7, 8, 9]
profile3.null_list # [0, 1, 2, 13, 14, 15, 26, 27, 28, 29] 
profile3.sample_size # 30
profile3.null_ratio # 0.3333

data4 = Data(file4)
#    [index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
#     [21, 22, 23, 24, 25, None, None, None, None]] 

profile3.update(data4) # initial null_list [6, 7, 8, 9], updated [16, 17, 18, 19]
profile3.null_list # [0, 1, 2, 13, 14, 15, 26, 27, 28, 29, 36, 37, 38, 39] <---- Is this accurate?
profile3.sample_size # 40
profile3.null_ratio # 0.35 
```

**Comment by ChrisWallace2020** [comment]
Solved with #199 in the integer case, the string case is much more complex, probably requires separate issue.
